includes/data-stores/class-wc-admin-reports-customers-data-store.php

```diff
@@ -505,6 +506,36 @@ public static function get_or_create_customer_from_order( $order ) {
 		return $result ? $customer_id : false;
 	}
 
+	/**
+	 * Get a customer name from billing, shipping, or user profile.
+	 *
+	 * @param WC_Order $order Order made by customer.
+	 * @return array
+	 */
+	public static function get_customer_name_from_order( $order ) {
```

Comment:
Should we add some unit test coverage here?

Reply:
Added some tests around the report to cover this.

Also updated the order to begin with profile first since that seems to be what's favored in core.